torch/testing/_internal/inductor_utils.py

```diff
@@ -49,3 +51,46 @@ def _check_has_dynamic_shape(
         has_dynamic, msg=f"Failed to find dynamic for loop variable\n{code}"
     )
     self.assertTrue(for_loop_found, f"Failed to find for loop\n{code}")
+
+@contextmanager
+def _temporary_log_handler(logger, handler, level):
+    logger.addHandler(handler)
+    prev_level = logger.level
+    logger.setLevel(level)
+    try:
+        yield
+    finally:
+        logger.setLevel(prev_level)
+        logger.removeHandler(handler)
+
+def run_and_get_code(fn, *args, **kwargs):
+    """
+    Run fn and return the result along with the generated code.
+    """
+    from torch._inductor import config
+
+    # We use the patch context manager instead of using it as a decorator.
+    # In this way, we can ensure that the attribute is patched and unpatched correctly
+    # even if this run_and_get_code function is called multiple times.
+    with patch.object(config, "debug", True):
```

Comment:
I know this was already the case with `run_and_get_cpp_code`, but it wasnt the case with `run_and_get_triton_code`. 

Debug does a lot of printing - can we avoid this ? I think it noticeably make tests slower.

Reply:
hmm, fair enough, let me try and run some benchmarks.

it looks like `run_and_get_triton_code` is doing a file read, which is not particularly fast either... but that file is probably in disk cache so I guess it might be okay